### PR TITLE
Unified documentation

### PR DESCRIPTION
**Pull request by garg-amit**
## Purpose

* Move all documents to the /docs directory.
* Remove any redundant docs(we can always revisit them if needed).
* Maintain easy switching b/w documents.
* Add comprehensive documentation to easily get started with federated learning on AML.

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[x] Documentation content changes
[ ] Other... Please describe:
```

### CODE REVIEW

File: docs/provisioning/internal_silos.md [hunk 1]
```diff
@@ -1,50 +0,0 @@
-# Provisioning a Federated Learning setup with internal silos
-
-## Contents
-
-This directory contains resources to help you provision a Federated Learning setup with _internal_ silos, _i.e._ silos that are in the same Azure tenant as the orchestrator. You will be able to use this setup to run the examples in the `../../examples/pipelines` directory.
-
-For this first iteration, we are only provisioning a _vanilla_ setup, _i.e._ a setup with no security or governance features, where the silos are NOT locked down. We will add these features in future iterations. In the mean time, you should NOT be uploading sensitive data to this setup. However, you CAN use this setup for refining your training pipelines and algorithms. The only elements that will need to change when working on a _real_ secure setup will just be the orchestrator and silos names in a config file - you will be able to re-use all your code as-is.
```
Comment (jfomhover) [inline docs/provisioning/internal_silos.md:7]: do we want to save this paragraph for another page?

File: docs/proposal.md [hunk 2]
```diff
@@ -1,74 +0,0 @@
-# Proposal for a Federated Learning Solution between Contoso and DataProvider 
-
-## Summary
-
-This document outlines a Federated Learning (FL) solution for data sharing between DataProvider and Contoso. We propose a plan that will enable leveraging DataProvider's data for improving Contoso's model accuracy without compromising any private data. The partnership will utilize DataProvider's data to adapt Contoso's pretrained model to a new domain (healthcare, finance...). Additionally, Contoso plans to fine-tune the service model based on the data that the company provides. Both training settings will utilize FL. In this document we suggest an architecture and implementation guidelines. 
-
-## Motivation
-
-Due to privacy regulations and limitations created due to trust boundaries, not all data used for training Machine Learning (ML) models can be gathered in a central location to train a model on all available data (_e.g._, some highly sensitive user data located in the DataProvider embassies). This constraint makes it hard to leverage the data for training models in classic training techniques.
```
Comment (jfomhover) [inline docs/proposal.md:9]: do we want to keep this paragraph in another page? motivations maybe?

File: docs/proposal.md [hunk 3]
```diff
@@ -1,74 +0,0 @@
...
-
-## Motivation
-
-Due to privacy regulations and limitations created due to trust boundaries, not all data used for training Machine Learning (ML) models can be gathered in a central location to train a model on all available data (_e.g._, some highly sensitive user data located in the DataProvider embassies). This constraint makes it hard to leverage the data for training models in classic training techniques. 
-
-An ML technique called Federated Learning (FL) does just that: train the various models locally on the data available in the clients (_a.k.a._ the silos, and in our case _embassies_), then combine the different models on a central server (_a.k.a._ the _orchestrator_) without data ever leaving the embassy.
```
Comment (jfomhover) [inline docs/proposal.md:11]: same comment, saving this?

File: docs/milestones.md [hunk 4]
```diff
@@ -1,99 +0,0 @@
...
-We describe the key milestones in detail below. 
-
-
-## Machine learning model 
-
-The prerequisite of a _federated_ learning job is a _machine_ learning model, which can be trained in a non-federated scheme. Such a model can be  
-- a working model trained with some real yet small data, or  
-- a model architecture that is proven effective via synthetic training data. 
-
-One example of a hospital (i.e., the customer) is described below. The hospital’s task is to classify a medical image into positive or negative of a specific disease. The data scientists at the hospital _already_ have a neural network model trained in a centralized fashion with their own patient data. The model achieved reasonable performance. Now the hospital wants to further improve the model's performance as more hospitals would like to participate in a federated learning way without their data leaving their clusters.
```
Comment (jfomhover) [inline docs/milestones.md:62]: keep it for a page explaining why FL is good for you?

**Review by jfomhover — APPROVED** [review]
Here's a couple suggestions of paragraphs we might want to keep. Use these comments as you see fit.

File: docs/proposal.md (hunk 2 above)
Comment (garg-amit) [reply docs/proposal.md:9]: Added it here https://github.com/Azure-Samples/azure-ml-federated-learning/pull/72/files#diff-0b5ca119d2be595aa307d34512d9679e49186307ef94201e4b3dfa079aa89938R13

File: docs/proposal.md (hunk 3 above)
Comment (garg-amit) [reply docs/proposal.md:11]: Added it here https://github.com/Azure-Samples/azure-ml-federated-learning/pull/72/files#diff-0b5ca119d2be595aa307d34512d9679e49186307ef94201e4b3dfa079aa89938R13

File: docs/milestones.md (hunk 4 above)
Comment (garg-amit) [reply docs/milestones.md:62]: Added it here https://github.com/Azure-Samples/azure-ml-federated-learning/pull/72/files#diff-0b5ca119d2be595aa307d34512d9679e49186307ef94201e4b3dfa079aa89938R36

File: docs/provisioning/internal_silos.md (hunk 1 above)
Comment (garg-amit) [reply docs/provisioning/internal_silos.md:7]: sure, here you go https://github.com/Azure-Samples/azure-ml-federated-learning/pull/72/files#diff-988a5392cf907f7c5728b8ac90445facc7f835f3c00375764ec99769dcde2d5aR21

**Review by jfomhover — APPROVED** [review]

**Pull request merged**